accepted/2021/workloads/workload-installation.md

```diff
@@ -0,0 +1,337 @@
+
+The advertising manifest will be updated as part of other commands (such as workload install or workload update).  It will use the following process for each workload manifest:
+
+- Each workload manifest should be delivered in a NuGet package
+  - The package ID should be the following: `<ManifestID>.Manifest.<SdkFeatureBand>`.  For example, `Microsoft.NET.Android.Manifest.6.0.100`.
+  - The manifest payload (`WorkloadManifest.json`, `WorkloadManifest.targets`, and any other files to be included with the manifest) should be in a `data/` folder in the NuGet package.  This ensures that there's not confusion about whether a file in the root is part of the workload manifest or a file owned by NuGet.
+- Download the latest version of the workload manifest NuGet package using NuGet
+  - The package should be extracted into the NuGet [global-packages](https://docs.microsoft.com/en-us/nuget/Consume-Packages/managing-the-global-packages-and-cache-folders) folder.
+
+*Possible optimization: For some installers, it will be necessary to have a native installation package that includes the workload manifest.  For example, there will need to be an MSI which installs the manifest files to the `dotnet` folder in `Program Files`.  To avoid downloading the same manifest twice, we could have the IAL be involved in downloading the workload manifest package, and extracting the information in it as the advertising manifest.  However, this could add significant complexity, and the manifests should be small, so we won't plan to do this optimization initially.*
+
+To install updated manifests, the following process will be used (for each workload manifest):
+
+- Check if advertising manifest is newer than the currently installed version
+  - This requires a way to get the NuGet package version of the workload manifest package corresponding to the currently installed version.  Either we need to support semantic versions in the version field of the workload manifest schema, or we need to store the NuGet package version separately (ie in a text file).
+  - If the installed version of the workload manifest is greater than or equal to the advertised version, then it is already up-to-date, and we are done with the update process for this manifest
+- Tell IAL: Install workload manifest (given ID and version)
+
+## Workload installation process
+ 
+- Input: list of workloads to install
+- With rollback:
```

Comment:
What does rollback mean in this context?

Reply:
This means it's in a "transaction" where if an operation fails, then previous operations will roll back to the initial state.